tests/Tests.AzureAppConfiguration/FeatureManagementTests.cs

```diff
@@ -866,5 +876,119 @@ public void SelectAndRefreshSingleFeatureFlag()
             Assert.Equal("Edge", config["FeatureManagement:Feature1:EnabledFor:0:Parameters:AllowedBrowsers:1"]);
         }
+
+        [Fact]
+        public void ValidateCorrectFeatureFlagLoggedDuringRefresh()
+        {
+            IConfigurationRefresher refresher = null;
+            var featureFlags = new List<ConfigurationSetting> { _kv2 };
+
+            var mockClient = new Mock<ConfigurationClient>(MockBehavior.Strict);
+
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(featureFlags));
+
+            var mockLogger = new Mock<ILogger>();
+            var mockLoggerFactory = new Mock<ILoggerFactory>();
+            mockLoggerFactory.Setup(mlf => mlf.CreateLogger(LoggingConstants.AppConfigRefreshLogCategory)).Returns(mockLogger.Object);
+
+            var mockClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.ClientManager = mockClientManager;
+                    options.UseFeatureFlags(o => o.CacheExpirationInterval = CacheExpirationTime);
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            Assert.Equal("SuperUsers", config["FeatureManagement:MyFeature2:EnabledFor:0:Name"]);
+
+            featureFlags[0] = ConfigurationModelFactory.ConfigurationSetting(
+            key: FeatureManagementConstants.FeatureFlagMarker + "myFeature1",
+            value: @"
+                    {
+                      ""id"": ""MyFeature"",
+                      ""description"": ""The new beta version of our web site."",
+                      ""display_name"": ""Beta Feature"",
+                      ""enabled"": true,
+                      ""conditions"": {
+                        ""client_filters"": [                        
+                          {
+                            ""name"": ""AllUsers""
+                          }
+                        ]
+                      }
+                    }
+                    ",
+            label: "label2",
+            contentType: FeatureManagementConstants.FeatureFlagContentType + ";charset=utf-8",
+            eTag: new ETag("c3c231fd-39a0-4cb6-3237-4614474b92c1" + "f"));
+
+            Thread.Sleep(CacheExpirationTime);
+            refresher.LoggerFactory = mockLoggerFactory.Object;
+            refresher.TryRefreshAsync().Wait();
+            Assert.Equal("AllUsers", config["FeatureManagement:MyFeature:EnabledFor:0:Name"]);
+            Assert.True(TestHelpers.ValidateLog(mockLogger, LoggingConstants.RefreshFeatureFlagChanged + "(key: '" + FeatureManagementConstants.FeatureFlagMarker + "myFeature2', label: '')", LogLevel.Debug));
```

Comment:
This test should be failing, I'm looking into why the build pipeline didn't report the failed test as pipeline failure. Until then, make sure to run tests locally.

Also, this test should've caught the bug that we are not logging on feature flag updates (only deletes are logged). I'll let you figure out why :)